pkg/authorization/decorator.go

```diff
@@ -126,3 +133,12 @@ func decisionString(dec authorizer.Decision) string {
 	}
 	return ""
 }
+
+// DelegateAuthorization delegates authorization to the given delegate authorizer
+// and prefixes the given reason with the reason after the given delegate authorizer executed.
+func DelegateAuthorization(delegationReason string, delegate authorizer.Authorizer) authorizer.Authorizer {
+	return authorizer.AuthorizerFunc(func(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+		dec, delegateReason, err := delegate.Authorize(ctx, attr)
+		return dec, "delegating due to " + delegationReason + ": " + delegateReason, err
```

Comment:
i'm not sure about `delegateReason` here, will it add more info or clutter the audit logs? Would the following also work?
```
"first/decision": "NoOpinion",
"second/reason":   "delegating to second due to abc",

"second/decision": "Denied",
"second/reason":   "failed due to def",
```

Reply:
@p0lyn0mial the problem with your suggestion is that `to second` will be infixed, potentially with a very long reason of the delegate.

Reply:
`second` is the name of the next delegate, `abc` should contain the reason of delegation